src/node_i18n.cc

```diff
@@ -501,19 +500,28 @@ void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
         converter->set_bom_seen(true);
       }
     }
-    ret = ToBufferEndian(env, &result);
-    if (omit_initial_bom && !ret.IsEmpty()) {
+
+    Local<Value> error;
+    const UChar* output = result.out();
+    size_t beginning = 0;
+    size_t length = result.length() * sizeof(UChar);
+
+    if (omit_initial_bom) {
       // Perform `ret = ret.slice(2)`.
-      CHECK(ret.ToLocalChecked()->IsUint8Array());
-      Local<Uint8Array> orig_ret = ret.ToLocalChecked().As<Uint8Array>();
-      ret = Buffer::New(env,
-                        orig_ret->Buffer(),
-                        orig_ret->ByteOffset() + 2,
-                        orig_ret->ByteLength() - 2)
-                            .FromMaybe(Local<Uint8Array>());
+      beginning += 2;
+      length -= 2;
     }
-    if (!ret.IsEmpty())
-      args.GetReturnValue().Set(ret.ToLocalChecked());
+
+    const char* value = reinterpret_cast<const char*>(output) + beginning;
+    MaybeLocal<Value> encoded =
+        StringBytes::Encode(env->isolate(), value, length, UCS2, &error);
+
+    if (!encoded.IsEmpty()) {
+      args.GetReturnValue().Set(encoded.ToLocalChecked());
+    } else {
+      args.GetReturnValue().Set(error);
+    }
```

Comment:
Just a nit... but to fit with the more common convention avoiding `ToLocalChecked`, this could be:

```suggestion
    v8::Local<v8::Value> ret;
    if (encoded.ToLocal(&ret)) {
      args.GetReturnValue().Set(ret);
    } else {
      args.GetReturnValue().Set(error);
    }
```

Reply:
I missed this and I'm just updating my last commit with the changes. Thanks for the review @jasnell